### ISSUE TITLE
Compilation not working

### ISSUE THREAD

**Issue by jebej** [andrewrk/libsoundio#142]
On elementaryOS, which is based off Ubuntu 16.04, following the readme instructions leads to errors during compilation:

```
jeremy@Newton-elementary:~/Documents/libsoundio-1.1.0/build$ cmake ..
Configuring libsoundio version 1.1.0
-- Could NOT find PULSEAUDIO (missing:  PULSEAUDIO_LIBRARY PULSEAUDIO_INCLUDE_DIR) 
-- Could NOT find ALSA (missing:  ALSA_LIBRARY ALSA_INCLUDE_DIR) 
-- Could NOT find COREAUDIO (missing:  COREAUDIO_LIBRARY COREAUDIO_INCLUDE_DIR) 
-- Could NOT find WASAPI (missing:  WASAPI_INCLUDE_DIR) 

Installation Summary
--------------------
* Install Directory            : /usr/local
* Build Type                   : Debug
* Build static libs            : ON
* Build examples               : ON
* Build tests                  : ON

System Dependencies
-------------------
* threads                      : OK
* JACK       (optional)        : OK
* PulseAudio (optional)        : not found
* ALSA       (optional)        : not found
* CoreAudio  (optional)        : not found
* WASAPI     (optional)        : not found

-- Configuring done
-- Generating done
-- Build files have been written to: /home/jeremy/Documents/libsoundio-1.1.0/build
jeremy@Newton-elementary:~/Documents/libsoundio-1.1.0/build$ make
[  2%] Building C object CMakeFiles/libsoundio_shared.dir/src/soundio.c.o
In file included from /home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:8:0:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio_private.h:112:49: error: unknown type name ‘SoundIoListDevicePtr’
 SOUNDIO_MAKE_LIST_STRUCT(struct SoundIoDevice*, SoundIoListDevicePtr, SOUNDIO_LIST_NOT_STATIC)
                                                 ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio_private.h:112:71: error: unknown type name ‘SOUNDIO_LIST_NOT_STATIC’
 SOUNDIO_MAKE_LIST_STRUCT(struct SoundIoDevice*, SoundIoListDevicePtr, SOUNDIO_LIST_NOT_STATIC)
                                                                       ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio_private.h:121:2: error: ISO C does not allow extra ‘;’ outside of a function [-Werror=pedantic]
 };
  ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio_private.h:168:57: error: unknown type name ‘SoundIoListSampleRateRange’
 SOUNDIO_MAKE_LIST_STRUCT(struct SoundIoSampleRateRange, SoundIoListSampleRateRange, SOUNDIO_LIST_NOT_STATIC)
                                                         ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio_private.h:168:85: error: unknown type name ‘SOUNDIO_LIST_NOT_STATIC’
 SOUNDIO_MAKE_LIST_STRUCT(struct SoundIoSampleRateRange, SoundIoListSampleRateRange, SOUNDIO_LIST_NOT_STATIC)
                                                                                     ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio_private.h:178:2: error: ISO C does not allow extra ‘;’ outside of a function [-Werror=pedantic]
 };
  ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:73:46: error: unknown type name ‘SoundIoListDevicePtr’
 SOUNDIO_MAKE_LIST_DEF(struct SoundIoDevice*, SoundIoListDevicePtr, SOUNDIO_LIST_NOT_STATIC)
                                              ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:73:68: error: unknown type name ‘SOUNDIO_LIST_NOT_STATIC’
 SOUNDIO_MAKE_LIST_DEF(struct SoundIoDevice*, SoundIoListDevicePtr, SOUNDIO_LIST_NOT_STATIC)
                                                                    ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c: In function ‘soundio_input_device_count’:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:302:33: error: dereferencing pointer to incomplete type ‘struct SoundIoDevicesInfo’
     return si->safe_devices_info->input_devices.length;
                                 ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c: In function ‘soundio_get_input_device’:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:363:36: error: implicit declaration of function ‘SoundIoListDevicePtr_val_at’ [-Werror=implicit-function-declaration]
     struct SoundIoDevice *device = SoundIoListDevicePtr_val_at(&si->safe_devices_info->input_devices, index);
                                    ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c: In function ‘soundio_device_unref’:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:398:16: error: dereferencing pointer to incomplete type ‘struct SoundIoDevicePrivate’
         if (dev->destruct)
                ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:409:9: error: implicit declaration of function ‘SoundIoListSampleRateRange_deinit’ [-Werror=implicit-function-declaration]
         SoundIoListSampleRateRange_deinit(&dev->sample_rates);
         ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c: In function ‘soundio_destroy_devices_info’:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:684:5: error: implicit declaration of function ‘SoundIoListDevicePtr_deinit’ [-Werror=implicit-function-declaration]
     SoundIoListDevicePtr_deinit(&devices_info->input_devices);
     ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c: In function ‘soundio_input_device_count’:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:303:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c: In function ‘soundio_output_device_count’:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:317:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c: In function ‘soundio_default_input_device_index’:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:331:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c: In function ‘soundio_default_output_device_index’:
/home/jeremy/Documents/libsoundio-1.1.0/src/soundio.c:345:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
cc1: all warnings being treated as errors
CMakeFiles/libsoundio_shared.dir/build.make:62: recipe for target 'CMakeFiles/libsoundio_shared.dir/src/soundio.c.o' failed
make[2]: *** [CMakeFiles/libsoundio_shared.dir/src/soundio.c.o] Error 1
CMakeFiles/Makefile2:173: recipe for target 'CMakeFiles/libsoundio_shared.dir/all' failed
make[1]: *** [CMakeFiles/libsoundio_shared.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by andrewrk** [comment]
I'm not able to reproduce this problem. What compiler version are you using?

**Comment by jebej** [comment]
I'm assuming I'm using gcc. In CMakeOutput.log, I see:
```
The system is: Linux - 4.4.0-51-generic - x86_64
Compiling the C compiler identification source file "CMakeCCompilerId.c" succeeded.
Compiler: /usr/bin/cc 
Build flags: 
Id flags: 

The output was:
0


Compilation of the C compiler identification source "CMakeCCompilerId.c" produced "a.out"

The C compiler identification is GNU, found in "/home/jeremy/Documents/libsoundio-1.1.0/build/CMakeFiles/3.5.1/CompilerIdC/a.out"
```

My gcc version is:
```
$ gcc --version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
```

**Comment by diogocp** [comment]
Looks like #67. The fix is in e8b908243d58760d7815525d18bddd64ec97a5d2.

**Comment by andrewrk** [comment]
I think @diogocp is right. @jebej please comment if building from master branch does not fix this problem for you. I'm working on getting out a new 1.x series release as well as a 2.x series release.
